build/result.go

```diff
@@ -192,32 +216,25 @@ func (r *ResultContext) getProcessConfig(cfg *controllerapi.InvokeConfig, stdin
 }
 
 func containerConfigFromResult(ctx context.Context, res *gateway.Result, c gateway.Client, cfg controllerapi.InvokeConfig) (*gateway.NewContainerRequest, error) {
-	if res.Ref == nil {
-		return nil, errors.Errorf("no reference is registered")
-	}
 	if cfg.Initial {
 		return nil, errors.Errorf("starting from the container from the initial state of the step is supported only on the failed steps")
 	}
-	st, err := res.Ref.ToState()
-	if err != nil {
-		return nil, err
-	}
-	def, err := st.Marshal(ctx)
+
+	ps, err := exptypes.ParsePlatforms(res.Metadata)
 	if err != nil {
 		return nil, err
 	}
-	imgRef, err := c.Solve(ctx, gateway.SolveRequest{
-		Definition: def.ToPB(),
-	})
```

Comment:
@ktock I'm not quite sure if I'm missing something here - I don't think this second evaluation is necessary?

Reply:
Yes, I think it's not necessary.